test/wallet-test.js

```diff
@@ -880,6 +880,75 @@ describe('Wallet', function() {
     assert.strictEqual(balance.unconfirmed, 5460);
   });
 
+  it('should fill tx with inputs using Branch and Bound', async () => {
+    const alice = await wdb.create();
+
+    // Coinbase
+    const t1 = new MTX();
+    t1.addInput(dummyInput());
+    t1.addOutput(await alice.receiveAddress(), 100000); // 99655
+    t1.addOutput(await alice.receiveAddress(), 50000); // 49655
+    t1.addOutput(await alice.receiveAddress(), 20000); // 19655
+    t1.addOutput(await alice.receiveAddress(), 10000); // 9655
+    t1.addOutput(await alice.receiveAddress(), 5000); // 4655
+    t1.addOutput(await alice.receiveAddress(), 1000); // 655
+
+    await wdb.addTX(t1.toTX());
+
+    // fee = 215
+    const outputs = [99440, 100095, 19440, 114405, 179060, 178405, 4440, 33750, 84060, 14095, 440];
```

Comment:
In "test: added tests for new selection algorithm" (f68b0a04ccd69d40133f55d626b60440200eab79):

I think all of these changeless transactions have an excess of exactly 0. I would encourage you to add also some values that test the upper bound of the changeless-solution window.

Reply:
Wow, this is great feedback. I should definitely add tests for the upper bound. Thanks!!

Reply:
It might be interesting to test a few different feerates in this context too!